frontend/src/individual_add.tsx

```diff
@@ -187,59 +200,97 @@ export function IndividualAdd({
+            litter: individual.litter,
+            id: null,
+          };
+
+          // Get all the breedings for the individual's origin herd and see if there is a match
+          const breedings: { breedings: Breeding[] } = await get(
+            `/api/breeding/${originHerd.herd}`
           );
+          const breedingMatch = breedings.breedings.find((item) => {
+            item.mother == currentBreeding.mother &&
+              item.father == currentBreeding.father &&
+              item.breed_date == currentBreeding.date;
+          });
+
+          // if there is no match, create a new breeding
+          if (!breedingMatch) {
+            birth = { ...birth, id: await createBreeding(currentBreeding) };
+          } else {
+            birth = { ...birth, id: breedingMatch.id };
+          }
+
+          // only continue with registering a birth if there is a breeding
+          if (!!birth.id) {
+            const birthEvent = await post("/api/birth", birth);
+
+            if (
+              birthEvent.status == "success" ||
+              birthEvent.message.includes("Birth already registered.")
+            ) {
+              const individualWithBreeding: Individual = {
+                ...newIndividual,
+                breeding: birth.id,
+              };
+              createIndividual(individualWithBreeding);
+            } else if (
+              birthEvent.status == "error" &&
+              birthEvent.message == "Not logged in"
+            ) {
+              userMessage(
+                "Du är inte inloggad. Logga in och försök igen.",
+                "warning"
+              );
+            } else if (
+              birthEvent.status == "error" &&
+              birthEvent.message.includes("litter size" || "Litter size")
+            ) {
+              userMessage("Ange en kullstorlek större än null.", "warning");
+            } else if (
+              birthEvent.status == "error" &&
+              birthEvent.message == "Forbidden"
+            ) {
+              userMessage(
+                "Du har inte behörighet att lägga till den här individen.",
+                "error"
+              );
+            } else {
+              userMessage(
+                // register_birth in data_access.py requires a litter size.
+                // Thus, the field must be mandatory in the form. Okay for föreningen?
+                "Något gick fel. Kolla så att du har lagt till mor, far, kullstorlek och födelsedatum.",
+                "error"
+              );
+            }
+          } else {
+            userMessage(
+              "Något gick fel på grund av tekniska problem. Kontakta en administrator.",
+              "error"
+            );
+          }
         } else {
-          userMessage("Individens nummer har inget giltigt format.", "warning");
+          userMessage(
+            "Första delen i individens nummer motsvarar ingen besättning.",
+            "warning"
+          );
```

Comment:
I'm having a hard time keeping track of all the cases here, would it be possible to decrease the depth and have simple 
if cases that return instead?